File: test-suite/src/test/java/io/micronaut/email/docs/SendAttachmentServiceTest.java
package io.micronaut.email.docs;

import io.micronaut.email.BodyType;
import io.micronaut.email.Email;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertTrue;

@MicronautTest(startApplication = false)
public class SendAttachmentServiceTest {

    @Inject
    SendAttachmentService sendAttachmentService;

    @Inject
    MockEmailSender<?> emailSender;

    @Test
    void transactionalTextEmailIsCorrectlyBuilt() throws IOException {
        //when:
        sendAttachmentService.sendReport();
        //then:
        assertEquals(1, emailSender.getEmails().size());
        Email email = emailSender.getEmails().get(0);
        assertEquals("sender@example.com", email.getFrom().getEmail());
        assertNull(email.getFrom().getName());
        assertEquals(1, email.getTo().size());
        assertEquals("john@example.com", email.getTo().stream().findFirst().get().getEmail());
        assertNull(email.getTo().stream().findFirst().get().getName());
        assertNull(email.getCc());
        assertNull(email.getBcc());
        assertEquals("Monthly reports", email.getSubject());
        assertNotNull(email.getBody());
        assertTrue(email.getBody().get(BodyType.TEXT).isPresent());
        assertEquals("Attached Monthly reports", email.getBody().get(BodyType.TEXT).get());
        assertTrue(email.getBody().get(BodyType.HTML).isPresent());
        assertEquals("<html><body><strong>Attached Monthly reports</strong>.</body></html>", email.getBody().get(BodyType.HTML).get());
        assertNotNull(email.getAttachments());
        assertEquals("reports.xlsx", email.getAttachments().get(0).getFilename());
        assertEquals("application/vnd.openxmlformats-officedocument.spreadsheetml.sheet", email.getAttachments().get(0).getContentType());
        assertNotNull(email.getAttachments().get(0).getContent());
    }
}

File: test-suite-python/src/test/python/micronaut/email/docs/SendAttachmentServiceTest.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.email import BodyType
from micronaut.email.mock import MockEmailSender
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .SendAttachmentService import SendAttachmentService


@Property(name="mock.emailsender", value="true")
@Property(name="javamail.enabled", value="false")
@MicronautTest(startApplication=False)
class SendAttachmentServiceTest:
    send_attachment_service: Annotated[SendAttachmentService, Inject]
    email_sender: Annotated[MockEmailSender, Inject]

    @Test
    def transactional_text_email_is_correctly_built(self):
        # when:
        self.send_attachment_service.send_report()
        # then:
        assert 1 == self.email_sender.getEmails().size()
        email = self.email_sender.getEmails().get(0)
        assert "sender@example.com" == email.getFrom().getEmail()
        assert email.getFrom().getName() is None
        assert 1 == email.getTo().size()
        assert "john@example.com" == email.getTo().stream().findFirst().get().getEmail()
        assert email.getTo().stream().findFirst().get().getName() is None
        assert email.getCc() is None
        assert email.getBcc() is None
        assert "Monthly reports" == email.getSubject()
        assert email.getBody() is not None
        assert email.getBody().get(BodyType.TEXT).isPresent()
        assert "Attached Monthly reports" == email.getBody().get(BodyType.TEXT).get()
        assert email.getBody().get(BodyType.HTML).isPresent()
        assert "<html><body><strong>Attached Monthly reports</strong>.</body></html>" == email.getBody().get(BodyType.HTML).get()
        assert email.getAttachments() is not None
        assert "reports.xlsx" == email.getAttachments().get(0).getFilename()
        assert "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet" == email.getAttachments().get(0).getContentType()
        assert email.getAttachments().get(0).getContent() is not None
